server/src/domain/asset/response-dto/asset-response.dto.ts

```diff
@@ -116,9 +116,13 @@ export function mapAsset(entity: AssetEntity, options: AssetMapOptions = {}): As
     tags: entity.tags?.map(mapTag),
     people: peopleWithFaces(entity.faces),
     checksum: entity.checksum.toString('base64'),
-    stackParentId: entity.stackParentId,
-    stack: withStack ? entity.stack?.map((a) => mapAsset(a, { stripMetadata })) ?? undefined : undefined,
-    stackCount: entity.stack?.length ?? null,
+    stackParentId: withStack ? entity.stack?.primaryAssetId : undefined,
+    stack: withStack
+      ? entity.stack?.assets
+          .filter((a) => a.id !== entity.stack?.primaryAssetId)
+          .map((a) => mapAsset(a, { stripMetadata })) ?? undefined
```

Comment:
That has been there before, I know. But `?? undefined` doesn't make any sense, right?

Reply:
Again, on phone but I would expect that this value could also be null. This syntax will set any false values to undefined.

Reply:
This only sets null to undefined, everything else passes through as is.

Reply:
Sorry, yes, it's specifically nulls it checks for and converts those to undefined

Reply:
Looking over the code now I am at my PC, I think you are right and here we should actually never get null, so this is unnecessary.

Reply:
![image](https://github.com/immich-app/immich/assets/1313578/8262e95f-f12d-46eb-93dc-916a66d1312b)
Typescript appears to agree with this :stuck_out_tongue: